slang-api/src/main/java/com/sonarsource/slang/utils/SyntacticEquivalence.java

```diff
@@ -67,16 +68,13 @@ public static boolean areEquivalent(@Nullable Tree first, @Nullable Tree second)
       return ((IdentifierTree) first).name().equals(((IdentifierTree) second).name());
     } else if (first instanceof LiteralTree) {
       return ((LiteralTree) first).value().equals(((LiteralTree) second).value());
-    } else if (first instanceof NativeTree) {
-      if (!((NativeTree) first).nativeKind().equals(((NativeTree) second).nativeKind())) {
-        return false;
-      }
-    } else if (first instanceof BinaryExpressionTree) {
-      if (((BinaryExpressionTree) first).operator() != ((BinaryExpressionTree) second).operator()) {
-        return false;
-      }
-    } else if (first instanceof AssignmentExpressionTree
-      && (((AssignmentExpressionTree) first).operator() != ((AssignmentExpressionTree) second).operator())) {
+    } else if ((first instanceof NativeTree) && (!((NativeTree) first).nativeKind().equals(((NativeTree) second).nativeKind()))) {
+      return false;
+    } else if ((first instanceof UnaryExpressionTree) && ((UnaryExpressionTree) first).operator() != ((UnaryExpressionTree) second).operator()) {
+      return false;
+    } else if ((first instanceof BinaryExpressionTree) && (((BinaryExpressionTree) first).operator() != ((BinaryExpressionTree) second).operator())) {
+      return false;
+    } else if ((first instanceof AssignmentExpressionTree) && (((AssignmentExpressionTree) first).operator() != ((AssignmentExpressionTree) second).operator())) {
```

Comment:
I think this could be refactored a bit. You could put all this conditions in a function (for instance `hasDifferentFields(first, second)` ). You would then only have 1  branch for all these `return false` cases